3-UsingAPIs/Week1/homework/ex3-rollDie.js

```diff
@@ -11,48 +11,42 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/3-Usin
   explanation? Add your answer as a comment to be bottom of the file.
 ------------------------------------------------------------------------------*/
 
-// TODO Remove callback and return a promise
-function rollDie(callback) {
-  // Compute a random number of rolls (3-10) that the die MUST complete
-  const randomRollsToDo = Math.floor(Math.random() * 8) + 3;
-  console.log(`Die scheduled for ${randomRollsToDo} rolls...`);
-
-  const rollOnce = (roll) => {
-    // Compute a random die value for the current roll
+function rollDie() {
+  return new Promise((resolve, reject) => {
+    const randomRollsToDo = Math.floor(Math.random() * 8) + 3;
+    console.log(`Die scheduled for ${randomRollsToDo} rolls...`);
+    const rollOnce = (roll) => {
+      
     const value = Math.floor(Math.random() * 6) + 1;
     console.log(`Die value is now: ${value}`);
-
-    // Use callback to notify that the die rolled off the table after 6 rolls
+  
     if (roll > 6) {
-      // TODO replace "error" callback
-      callback(new Error('Oops... Die rolled off the table.'));
+      reject(new Error('Oops... Die rolled off the table.'));
     }
 
-    // Use callback to communicate the final die value once finished rolling
     if (roll === randomRollsToDo) {
-      // TODO replace "success" callback
-      callback(null, value);
-    }
-
-    // Schedule the next roll todo until no more rolls to do
-    if (roll < randomRollsToDo) {
-      setTimeout(() => rollOnce(roll + 1), 500);
+      
+      resolve(value);
     }
-  };
-
-  // Start the initial roll
-  rollOnce(1);
+  
+      // Schedule the next roll todo until no more rolls to do
+      if (roll < randomRollsToDo) {
+        setTimeout(() => rollOnce(roll + 1), 500);
+      }
+    };
+  
+    rollOnce(1);
+
+  })
+  
+
+  
 }
 
 function main() {
-  // TODO Refactor to use promise
-  rollDie((error, value) => {
-    if (error !== null) {
-      console.log(error.message);
-    } else {
-      console.log(`Success! Die settled on ${value}.`);
-    }
-  });
+  rollDie()
+  .then(value => console.log(`Success! Die settled on ${value}.`))
+  .catch(error => console.log(error.message))
```

Comment:
It is customary to indent chained methods like below. Again, this would normally done by Prettier.

```suggestion
  rollDie()
    .then(value => console.log(`Success! Die settled on ${value}.`))
    .catch(error => console.log(error.message))
```

Reply:
I have an issue with the shape of codes. I will fix it :)